lib/common/changer.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module ReleaseManager
+  module Common
+    class Changer
+      class << self
+        def change_promotion(request)
+          create_component
+          clone_ci_configs
+          git_helper.use_repo(CI_CONFIGS_DIR) do
+            modify_file(request)
+            commit_and_push_changes(request)
+          end
+        end
+
+        private
+
+        def create_component
+          @ci_job_configs = factory.create_component(file_name: CI_CONFIGS_DIR, url: CI_CONFIGS_URL, ref: '')
+        end
+
+        def clone_ci_configs
+          logger.info('Cloning ci-job-configs...')
+          cloner.clone_component(@ci_job_configs)
+        end
+
+        def modify_file(request)
+          file_path = CI_CONFIGS_DIR.join('jenkii/platform/projects/puppet-agent.yaml')
+          branch = request.source_branch.tr('.', '')
+          temp_file = Tempfile.new('foo')
+          found = false
+          File.open(file_path, 'r').each_line do |line|
+            if found == true
+              line = modify_promotion(line)
+              found = false
+            end
+            found = true if /&#{branch}_agent_pe_promotion/.match?(line)
+            temp_file.puts line
+          end
+          temp_file.close
+          FileUtils.mv(temp_file.path, file_path)
+        end
+
+        def commit_and_push_changes(request)
+          git_helper.commit("(maint) Disable PE promotion for #{request.source_branch}")
+          git_helper.push
+        end
+
+        def modify_promotion(line)
+          return line.gsub('TRUE', 'FALSE') if line.match?('TRUE')
```

Comment:
this should be /TRUE/.match?(line), since match? is a method called on a regexp.